universal_transfer_operator/transfers/data_providers/__init__.py

```diff
@@ -0,0 +1,43 @@
+import importlib
+import os.path
+
+from transfers.constants import TransferMode
+from transfers.data_providers.base import DataProviders, TempFile
+from transfers.datasets.base import UniversalDataset as Dataset
+
+from astro.constants import LoadExistStrategy
+from astro.utils.path import get_class_name
+
+DATASET_CONN_ID_TO_DATAPROVIDER_MAPPING = {
+    "s3": "data_providers.filesystem.aws.s3",
+    "aws": "data_providers.filesystem.aws.s3",
+    "gs": "data_providers.filesystem.google.cloud.gcs",
+    "google_cloud_platform": "data_providers.filesystem.google.cloud.gcs",
+}
+
+
+def create_dataprovider(
+    dataset: Dataset,
+    transfer_params: dict = {},
+    transfer_mode: TransferMode = TransferMode.NONNATIVE,
+    if_exists: LoadExistStrategy = "replace",
+) -> DataProviders:
+    from airflow.hooks.base import BaseHook
+
+    conn_type = BaseHook.get_connection(dataset.conn_id).conn_type
+    module_path = DATASET_CONN_ID_TO_DATAPROVIDER_MAPPING[conn_type]
+    module = importlib.import_module(module_path)
+    class_name = get_class_name(module_ref=module, suffix="DataProvider")
+    data_provider: DataProviders = getattr(module, class_name)(
+        dataset=dataset,
+        transfer_params=transfer_params,
+        transfer_mode=transfer_mode,
+        if_exists=if_exists,
+    )
+    return data_provider
+
+
+def cleanup(file_list: list[TempFile]):
+    for file in file_list:
+        if os.path.exists(file.tmp_file.name):
+            os.remove(file.tmp_file.name)
```

Comment:
Shouldn't the cleanup be part of data-provider too? as how to do cleanup will vary on the type of provider

Reply:
Added a context manager so that clean-up happens in `_exit__()`.  https://github.com/astronomer/astro-sdk/pull/1492/commits/831d627d1a86acab300968fc56d11bc7f016b4b0